File: common/src/main/java/com/xxm/mmd/common/data/AppDataBase.java
package com.xxm.mmd.common.data;

import android.arch.persistence.db.SupportSQLiteOpenHelper;
import android.arch.persistence.room.Database;
import android.arch.persistence.room.DatabaseConfiguration;
import android.arch.persistence.room.InvalidationTracker;
import android.arch.persistence.room.RoomDatabase;

import com.xxm.mmd.common.data.bean.UserBean;
import com.xxm.mmd.common.data.dao.UserDao;

/**
 * Created by MaDeng on 2018/5/4.
 */
@Database(entities = {UserBean.class},version = 1)
public abstract class AppDataBase extends RoomDatabase {
    public abstract UserDao getUserDao();
}

File: app/src/main/java/com/xxm/mmd/wxfxk/data/AppDataBase.kt
package com.xxm.mmd.wxfxk.data

import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context

/**
 * Created by MaDeng on 2018/4/19.
 */
abstract class AppDataBase :RoomDatabase(){

    companion object {
        @Volatile private var INSTANCE: AppDataBase? = null
        fun getInstance(context: Context):AppDataBase =
                INSTANCE ?: synchronized(this){
                    INSTANCE?:buildDataBase(context).also{ INSTANCE = it}
                }
        private fun buildDataBase(context: Context) =
                Room.databaseBuilder(context.applicationContext,
                        AppDataBase::class.java,"wxfx.db")
                        .build()
    }

}